src/lib/server/websearch/scrape/playwright.ts

```diff
@@ -0,0 +1,57 @@
+import {
+	type BrowserContext,
+	chromium,
+	devices,
+	type Page,
+	type BrowserContextOptions,
+} from "playwright";
+import { PlaywrightBlocker } from "@cliqz/adblocker-playwright";
+import { env } from "$env/dynamic/private";
+
+async function initPlaywrightService() {
+	if (playwrightService) return playwrightService;
+
+	const browser = await chromium.launch({ headless: true });
+
+	process.on("SIGINT", () => browser.close());
+
+	const device = devices["Desktop Chrome"];
+	const options: BrowserContextOptions = {
+		...device,
+		// Increasing width improves spatial clustering accuracy
+		screen: {
+			width: 3840,
+			height: 1080,
+		},
+		viewport: {
+			width: 3840,
+			height: 1080,
+		},
+		reducedMotion: "reduce",
+		acceptDownloads: false,
+		timezoneId: "America/New_York",
+		locale: "en-US",
+	};
+	const ctx = await browser.newContext(options);
+	const blocker = await PlaywrightBlocker.fromPrebuiltAdsAndTracking(fetch).then((blker) => {
+		const mostBlocked = blker.blockFonts().blockMedias().blockFrames().blockImages();
+		if (env.WEBSEARCH_DISABLE_JAVASCRIPT) return mostBlocked.blockScripts();
+		return mostBlocked;
+	});
+	return Object.freeze({ ctx, blocker });
+}
+let playwrightService: Promise<{ ctx: BrowserContext; blocker: PlaywrightBlocker }>;
```

Comment:
mybe put this line before `initPlaywrightService` function for the clarity and add a comment that it is a singleton